shared/js/views.jsx

```diff
@@ -636,6 +638,79 @@ loop.shared.views = (function(_, mozL10n) {
   });
 
   /**
+   * Renders a tile with information about the page that has been added/deleted from the ToC.
+   *
+   * @property {Boolean} allowClick         Set to true to allow the url to be clicked. If this
+   *                                        is specified, then 'dispatcher' is also required.
+   * @property {String}  description        The description for the context url.
+   * @property {loop.Dispatcher} dispatcher
+   * @property {String}  thumbnail          The thumbnail url (expected to be a data url) to
+   *                                        display. If not specified, a fallback url will be
+   *                                        shown.
+   * @property {String}  url                The url to be displayed. If not present or invalid,
+   *                                        then this view won't be displayed.
+   * @property {String}  username           The name of the user that modified the ToC page
+   *
+   */
+  var ChatPageTileView = React.createClass({
+    mixins: [React.addons.PureRenderMixin],
+
+    propTypes: {
+      allowClick: React.PropTypes.bool.isRequired,
+      description: React.PropTypes.string,
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher),
+      thumbnail: React.PropTypes.string,
+      url: React.PropTypes.string,
+      username: React.PropTypes.string
+    },
+
+    /**
+     * Dispatches an action to record when the link is clicked.
+     */
+    handleLinkClick: function() {
+      if (!this.props.allowClick) {
+        return;
+      }
+
+      this.props.dispatcher.dispatch(new sharedActions.RecordClick({
+        linkInfo: "Shared URL"
+      }));
+    },
+
+    render: function() {
+      var description = this.props.description || null;
+      var thumbnail = this.props.thumbnail;
+      var url = this.props.url || null;
+      var username = this.props.username || null;
+      var sanitizedURL = loop.shared.utils.formatSanitizedContextURL(url) || {};
+      var hostname = sanitizedURL.hostname || null;
```

Comment:
Move the `null` values to `getDefaultProps` function. That way, you don't need to check if a props is defined or not:

`getDefaultProps: function() {
    return {
        description: null,
        ...
    };
}`